analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NativeMethodsShouldBeWrapped.CSharp11.cs

```diff
@@ -8,3 +8,32 @@ public interface IInterface
     [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
     public static extern virtual bool RemoveDirectory2(string name);  // Noncompliant
 }
+
+namespace LibraryImportAttributeTests
+{
+    // Copy the class to a .Net 7 project in VS and use "goto definition" on the methods to extract the code generated by the Microsoft.Interop.LibraryImportGenerator
+    public static partial class ExternMethods
+    {
+        [LibraryImport("foo.dll")]
+        public static partial void DllImportAttributeAppliedToThisFunction(); // Noncompliant
+
+        [LibraryImport("foo.dll")]
+        private static partial void CompliantDllImportAttributeAppliedToThisFunction(int i); // Compliant
+
+        [LibraryImport("foo.dll", StringMarshalling = StringMarshalling.Utf8)]
+        public static partial void DllImportAttributeAppliedToGeneratedLocalFunction(string p); // Noncompliant
+
+        [LibraryImport("foo.dll", StringMarshalling = StringMarshalling.Utf8)]
+        private static partial void CompliantDllImportAttributeAppliedToGeneratedLocalFunction(string p); // Compliant
+
+        // Wrapper tests
+
+        public static void CompliantDllImportAttributeAppliedToThisFunctionWrapper(int i) // Noncompliant {{Make this wrapper for native method 'CompliantDllImportAttributeAppliedToThisFunction' less trivial.}}
```

Comment:
nitpick: can you also add a test to ensure that when it has a "non trivial" wrapper, it passes (it's a bit of duplication in tests, but it can be a good regression test)